### PR TITLE
Added version printing cmdline options

### PR DESCRIPTION
**Pull request by FrostyMike**
Added version tag, extending the original version number of the package. Additionally added cmdline options --version and --version-tag to F_Front and F_Back.

Background: company which produced XCodeMLTools no longer supports them, it is unlikely that there will ever be another version increase. Because of that introduced an optional tag (arbitrary string) which will be added to the version. In practise, that would be git commit's hash.

### CODE REVIEW

File: configure.in [hunk 1]
```diff
@@ -4,7 +4,35 @@ AC_CONFIG_AUX_DIR([./buildutils])
 AC_CONFIG_MACRO_DIR([./buildutils])
 AC_CANONICAL_TARGET
 AC_CONFIG_SRCDIR([include/exc_platform.h])
-AC_CONFIG_HEADER([include/config.h Driver/atool/src/include/nata/nata_config.h])
+AC_CONFIG_HEADER([include/config.h Driver/atool/src/include/nata/nata_config.h ])
```
Comment (elsagermann) [inline configure.in]: Is the trailing ws here on purpose?

File: configure.in [hunk 2]
```diff
@@ -4,7 +4,35 @@ AC_CONFIG_AUX_DIR([./buildutils])
...
+    local var="$*"
+    # remove leading whitespace characters
+    var="${var#"${var%%[![:space:]]*}"}"
+    # remove trailing whitespace characters
+    var="${var%"${var##*[![:space:]]}"}"
+    printf '%s' "$var"
+}
+#--------------------------------------------------------------------
+
+AC_ARG_WITH(version-tag,
```
Comment (elsagermann) [inline configure.in:22]: _version_tag_ not version-tag?

File: configure.in [hunk 3]
```diff
@@ -4,7 +4,35 @@ AC_CONFIG_AUX_DIR([./buildutils])
...
+	    [  --with-version-tag=<tag> Add arbitrary string tag to the xcodeml-tools version],
+		[with_version_tag_specified=yes], [with_version_tag_specified=no])
+
+if test "x${with_version_tag_specified}" = "xyes"; then
+  # Trim tag string
+  with_version_tag=$(trim "${with_version_tag}")
```
Comment (elsagermann) [inline configure.in:28]: This won't work otherwise?

File: configure.in (hunk 2 above)
Comment (elsagermann) [reply configure.in:22]: Nah. sorry, they are interchangeable since - is transformed to _

File: configure.in (hunk 1 above)
Comment (FrostyMike) [reply configure.in]: No, removed it.

File: configure.in (hunk 2 above)
Comment (FrostyMike) [reply configure.in:22]: No, actually should be "version-tag" cause this is a text of cmdline option. However, the variable which will be created to hold the value afterwards will be named "version_tag".

File: configure.in (hunk 3 above)
Comment (FrostyMike) [reply configure.in:28]: It will (technically) work without "trim", but I didn't want leading & trailing spaces to be part of the tag.

**Pull request merged**